nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,487 @@
+                          thresholding_strategy='percentile'):
+    """ A function which keeps the most prominent regions of the maps,
+    denoted as foreground objects/regions extraction.
+
+    Parameters
+    ----------
+    maps_img: a 3D/4D Nifti like image/object
+        a path or filename or an image consists of statistical maps or atlas
+        maps.
+
+    mask_img: a Nifti like image/object, default is None
+        If given, mask image is applied directly on the input data 'maps_img'.
+        For example, if interested to consider only one particular brain region
+        then mask image to that particular brain region should be provided.
+        If none, masking will not be applied.
+
+    threshold: a float (t-value or ratio [0., 100.]), 'auto', default is 'auto'
+        a value used to threshold the maps to keep the most meaningful voxels.
+
+        If float representing as a t-value, a t-statistic value should be submitted
+        to keep the voxel intensities which are above than this value.
+        Mostly suitable, if user wants to keep the voxels based on raw statistic value.
+        In this case, user should know exactly which are the voxels to be kept by
+        visually checking the intensity value.
+
+        If float representing as a ratio, this value is used to keep the more intense
+        voxels across all maps. The most meaningful voxels are kept by multiplying
+        this value with the total size of the voxels.
+
+        If default 'auto', a pre-defined float value set to 0.8 will be used
+        in representing as a ratio.
+
+    thresholding_strategy: string {None, 'ratio_n_voxels', 'percentile'}, \
+        default 'percentile'.
+        a string used to select between two different types of thresholding strategies.
+        This strategy takes the given threshold value and thresholds the data.
+
+        If ratio_n_voxels, most meaningful voxels which are survived above the
+        value which is determined by ratio * n_voxels will be kept.
+
+        If percentile, most meaningful voxels which are survived above the
+        value determined by (percentile = ratio * 100) of the total voxels.
+
+        If None, neither of the strategy is used which means no thresholding will
+        be done.
+
+    Returns
+    -------
+    threshold_maps_img: a Nifti like image/object
+        a thresholded image of the input.
+    """
+    maps_img = check_niimg(maps_img, atleast_4d=True)
+    maps = maps_img.get_data()
+    len_of_maps = maps.shape[-1]
+    affine = maps_img.get_affine()
+
+    if mask_img is not None:
+        if not _check_same_fov(maps_img, mask_img):
+            mask_img = resample_img(mask_img,
+                                    target_affine=maps_img.get_affine(),
+                                    target_shape=maps_img.shape[:3],
+                                    interpolation="nearest")
+
+        mask = masking._load_mask_img(mask_img)
```

Comment:
`_load_mask_img` is an exception in nilearn. It takes an niimg as input and returns a tuple of (data, affine). This is because the type used to store niimg (nibabel.Nifti1Image) does not support boolean types. You should therefore write `mask, _ = masking._load_mask_img(mask_img)`